test/exercise/fp2/solution.rb

```diff
@@ -1,23 +1,37 @@
 module Exercise
   module Fp2
     class MyArray < Array
-      # Использовать стандартные функции массива для решения задач нельзя.
-      # Использовать свои написанные функции для реализации следующих - можно.
-
-      # Написать свою функцию my_each
       def my_each
+        for element in self
+          yield element
+        end
       end
 
-      # Написать свою функцию my_map
       def my_map
+        new_array = MyArray.new
+        self.my_each do |element|
+          new_array.push(yield element)
+        end
+        new_array
       end
 
-      # Написать свою функцию my_compact
       def my_compact
+        new_array = MyArray.new
+        self.my_each do |element|
+          !element.nil? && new_array.push(element)
```

Comment:
Try to overwrite it with a single-line `if/unless`.

Reply:
that line was removed by newest implementation